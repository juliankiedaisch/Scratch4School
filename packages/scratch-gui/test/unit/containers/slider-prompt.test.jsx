import React from 'react';
import SliderPrompt from '../../../src/containers/slider-prompt.jsx';
import {screen, fireEvent} from '@testing-library/react';
import {renderWithIntl} from '../../helpers/intl-helpers.jsx';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';


describe('Slider Prompt Container', () => {

    const store = configureStore()({
        locales: {
            isRtl: false,
            locale: 'en-US'
        }
    });


    let onCancel;
    let onOk;

    beforeEach(() => {
        onCancel = jest.fn();
        onOk = jest.fn();
    });

    test('matches snapshot', () => {
        renderWithIntl(
            <Provider store={store}>
                <SliderPrompt
                    isDiscrete={false}
                    maxValue={100}
                    minValue={0}
                    onCancel={onCancel}
                    onOk={onOk}
                />
            </Provider>
        );
        expect(document.body).toMatchSnapshot();
    });

    test('calls onOk when confirm button is clicked', () => {
        renderWithIntl(
            <Provider store={store}>
                <SliderPrompt
                    isDiscrete={false}
                    maxValue={100}
                    minValue={0}
                    onCancel={onCancel}
                    onOk={onOk}
                />
            </Provider>
        );

        fireEvent.click(screen.getByRole('button', {name: 'OK'}));
        expect(onOk).toHaveBeenCalled();
    });

    test('calls onCancel when cancel button is clicked', () => {
        renderWithIntl(
            <Provider store={store}>
                <SliderPrompt
                    isDiscrete={false}
                    maxValue={100}
                    minValue={0}
                    onCancel={onCancel}
                    onOk={onOk}
                />
            </Provider>
        );

        fireEvent.click(screen.getByRole('button', {name: 'Cancel'}));
        expect(onCancel).toHaveBeenCalled();
    });
});
