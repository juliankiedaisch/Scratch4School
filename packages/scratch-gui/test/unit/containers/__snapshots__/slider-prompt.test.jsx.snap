// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Prompt Container matches snapshot 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75);"
    >
      <div
        aria-label="Change slider range"
        aria-modal="true"
        class="ReactModal__Content scratch-editor-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class=""
          dir="ltr"
          style="flex-direction: column; flex-grow: 1;"
        >
          <div
            class=""
          >
            <div
              class=""
            >
              Change slider range
            </div>
            <div
              class=""
            >
              <div
                aria-label="Close"
                class=""
                role="button"
                tabindex="0"
              >
                <img
                  class=""
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <div
              class=""
            >
              Minimum value
            </div>
            <div
              class=""
            >
              <input
                name="Minimum value"
                pattern="-?[0-9]*(\\.[0-9]+)?"
                type="text"
                value="0.00"
              />
            </div>
            <div
              class=""
            >
              Maximum value
            </div>
            <div
              class=""
            >
              <input
                name="Maximum value"
                pattern="-?[0-9]*(\\.[0-9]+)?"
                type="text"
                value="100.00"
              />
            </div>
            <div
              class=""
            >
              <button>
                Cancel
              </button>
              <button>
                OK
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
